client-app/shared/catalog/composables/useProductVariationProperties.test.ts

```diff
@@ -0,0 +1,342 @@
+    const variationsWithAutoSelect = ref(mockData.autoSelect);
+    const { select, isSelected } = useProductVariationProperties(variationsWithAutoSelect);
+
+    select("Color", "Red");
+    expect(isSelected("Fit", "Slim")).toBe(true);
+  });
+
+  it("resets incompatible selections when a new property is selected", () => {
+    const variations = ref(mockData.basic);
+    const { select, isSelected } = useProductVariationProperties(variations);
+
+    select("Size", "L");
+    expect(isSelected("Size", "L")).toBe(true);
+    select("Color", "Blue");
+    expect(isSelected("Size", "L")).toBe(false);
+    expect(isSelected("Color", "Blue")).toBe(true);
+    expect(isSelected("Size", "M")).toBe(true); // auto-selected
+  });
+
+  it("correctly identifies when all properties are selected (isCompleted)", () => {
+    const variations = ref(mockData.basic.slice(0, 3));
+    const { select, isCompleted } = useProductVariationProperties(variations);
+
+    expect(isCompleted.value).toBe(false);
+    select("Color", "Red");
+    expect(isCompleted.value).toBe(false);
+    select("Size", "M");
+    expect(isCompleted.value).toBe(true);
+  });
+
+  it("returns the correct final variation when selection is complete", () => {
+    const variations = ref(mockData.basic.slice(0, 3));
+    const { select, variationResult } = useProductVariationProperties(variations);
+
+    select("Color", "Red");
+    select("Size", "M");
+    expect(variationResult.value).toBeDefined();
+    expect(variationResult.value?.id).toBe("1");
+  });
+
+  it("returns undefined for variationResult when selection is not complete", () => {
+    const variations = ref(mockData.basic.slice(0, 3));
+    const { select, variationResult } = useProductVariationProperties(variations);
+
+    select("Color", "Red");
+    expect(variationResult.value).toBeUndefined();
+  });
+
+  it("handles variations with different property counts", () => {
+    const variations = ref(mockData.basic);
+    const { select, isSelected, isAvailable } = useProductVariationProperties(variations);
+
+    select("Material", "Cotton");
+    expect(isSelected("Material", "Cotton")).toBe(true);
+    expect(isAvailable("Color", "Red")).toBe(false);
+    expect(isAvailable("Size", "M")).toBe(false);
+  });
+
+  it("resets selections when the variations array changes", async () => {
+    const variations = ref(mockData.basic.slice(0, 1));
+    const { select, isSelected, variationResult } = useProductVariationProperties(variations);
+
+    select("Color", "Red");
+    select("Size", "M");
+    expect(isSelected("Color", "Red")).toBe(true);
+    expect(variationResult.value).toBeDefined();
+
+    variations.value = mockData.basic.slice(2, 3);
+    await nextTick();
+
+    expect(isSelected("Color", "Red")).toBe(false);
+    expect(variationResult.value).toBeUndefined();
+  });
+
+  it("handles an empty variations array", () => {
+    const variations = ref<readonly Product[]>([]);
```

Comment:
Try without generic. Lets follow recommendations
<img width="817" height="124" alt="image" src="https://github.com/user-attachments/assets/6f165bef-0382-4289-b502-f60bd29e0a67" />

Reply:
or if needed just put it inside [] as readonly...